src/__tests__/proposals.int.test.ts

```diff
@@ -146,6 +146,115 @@ describe('/proposals', () => {
     });
   });
 
+  describe('GET /:id', () => {
+    it('returns 404 when given id is not present', async () => {
+      await agent
+        .get('/proposals/9001')
+        .set(dummyApiKey)
+        .expect(404, { message: 'Not found. Find existing proposals by calling with no parameters.' });
+    });
+
+    it('returns the one proposal asked for', async () => {
+      await db.query(`
+        INSERT INTO opportunities (
+          title,
+          created_at
+        )
+        VALUES
+          ( '⛰️', '2525-01-03T00:00:01Z' )
+      `);
+      await db.query(`
+        INSERT INTO applicants (
+          external_id,
+          opted_in,
+          created_at
+        )
+        VALUES
+          ( '🐕', 'true', '2525-01-03T00:00:02Z' ),
+          ( '🐈', 'false', '2525-01-03T00:00:03Z' );
+      `);
+      await db.query(`
+        INSERT INTO proposals (
+          applicant_id,
+          external_id,
+          opportunity_id,
+          created_at
+        )
+        VALUES
+          ( 1, 'proposal-1', 1, '2525-01-03T00:00:04Z' ),
+          ( 1, 'proposal-2', 1, '2525-01-03T00:00:05Z' );
+      `);
+      await agent
+        .get('/proposals/2')
+        .set(dummyApiKey)
+        .expect(
+          200,
+          {
+            id: 2,
+            externalId: 'proposal-2',
+            applicantId: 1,
+            opportunityId: 1,
+            createdAt: '2525-01-03T00:00:05.000Z',
+          },
+        );
+    });
+
+    it('should error if the database returns an unexpected data structure', async () => {
+      jest.spyOn(db, 'sql')
+        .mockImplementationOnce(async () => ({
+          rows: [{ foo: 'not a valid result' }],
+        }) as Result<object>);
+      const result = await agent
+        .get('/proposals/2')
+        .set(dummyApiKey)
+        .expect(500);
+      expect(result.body).toMatchObject({
+        name: 'InternalValidationError',
+        details: expect.any(Array) as unknown[],
```

Comment:
Dropping a pin here because if [my other comment](https://github.com/PhilanthropyDataCommons/service/pull/192/files#r1065075986) ends up having legs, then I would expect this test is only passing because we're always falling back to the empty array, and need to actually check that whatever `isProposal.errors` is expected to return is actually being returned.

If that comment _doesn't_ have legs, this one can be dismissed too.

Reply:
I'm not sure I fully understand the suggestion but I know this test can pass when you call the wrong `isWhatever.errors` because we're not actually looking at the exact expected error message, so I agree there is some improvement to be had here.

As for it being an array, I think it's always going to be an array due to the `ValidationError` type defined in `src/errors/ValidationError.ts` and used by `src/middleware/errorHandler.ts` to structure the response.

Reply:
What I meant was, if it turned out that `isProposal.errors` _was_ a broken invocation (which turns out not to be the case), and was always failing / falling back to `[]`, then we wouldn't be catching that, and probably should be. But that's not the case so this comment can be resolved.

But yeah, to your point, we should probably check to make sure an expected error is returned.

Reply:
Oh, I see, back in the handler code that fallback to empty array. I think I got it.

Reply:
Right! Like as-is, this test wouldn't catch if we had accidentally typed `isProposal.erors || []` back in the handler (note the `erors` typo). Instead we'd just… never be returning the errors. Dunno if that's a useful addition of test coverage or not tho.